src/components/Character.js

```diff
@@ -1,10 +1,36 @@
 import "./character.css";
 import { createElement } from "../utils/elements";
+import FaveActive from "../assets/star-active.svg";
+import FaveInactive from "../assets/star-inactive.svg";
 
 function createCharacter({ ...props }) {
+  const favoriteImg = createElement("img", {
+    src: props.isFavorite ? FaveActive : FaveInactive,
+  });
+  const favoriteBtn = createElement("button", {
+    className: "faveBtn",
+    children: [favoriteImg],
+    onclick: () => {
+      let currentFavorites = JSON.parse(
+        localStorage.getItem("favorites") || "[]"
+      );
+      const isFavorite = currentFavorites.includes(name);
+      if (isFavorite) {
+        currentFavorites = currentFavorites.filter(
+          (favorite) => favorite !== name
+        );
+      } else {
+        currentFavorites.push(name);
+      }
+      localStorage.setItem("favorites", JSON.stringify(currentFavorites));
+      favoriteImg.src = !isFavorite ? FaveActive : FaveInactive;
+    },
+  });
```

Comment:
```suggestion
      const isFavorite = currentFavorites.includes(props.name);
      if (isFavorite) {
        currentFavorites = currentFavorites.filter(
          (favorite) => favorite !== props.name
        );
      } else {
        currentFavorites.push(props.name);
      }
      localStorage.setItem("favorites", JSON.stringify(currentFavorites));
      favoriteImg.src = !isFavorite ? FaveActive : FaveInactive;
    },
  });
```

The local storage should work if you add the props to every name element 🙌